lib/web_ui/dev/integration_tests_manager.dart

```diff
@@ -0,0 +1,345 @@
+        ['--port=4444'],
+        workingDirectory: pathlib.join(
+            _driverDir.path, 'web_installers', 'packages', 'web_drivers'),
+      );
+      print('INFO: Driver started');
+      return true;
+    }
+  }
+
+  Future<bool> prepareDriver() async {
+    if (_driverDir.existsSync()) {
+      _driverDir.deleteSync(recursive: true);
+    }
+
+    _driverDir.createSync(recursive: true);
+
+    bool installWebInstallers = await _cloneWebInstalllers();
+    if (installWebInstallers) {
+      bool pubGet = await _runPubGet(pathlib.join(
+          _driverDir.path, 'web_installers', 'packages', 'web_drivers'));
+      if (pubGet) {
+        return await _runDriver();
+      } else {
+        return false;
+      }
+    }
+    return false;
+  }
+
+  /// Runs the all the web tests under e2e_tests/web.
+  Future<bool> _runTests() async {
+    // Only list the files under e2e_tests/web.
+    final List<io.FileSystemEntity> entities =
+        environment.integrationTestsDir.listSync(followLinks: false);
+
+    print('INFO: Listing test files under directory: '
+        '${environment.integrationTestsDir.path.toString()}');
+    bool testResults = true;
+    for (io.FileSystemEntity e in entities) {
+      // The tests should be under this directories.
+      if (e is io.Directory) {
+        testResults = testResults && await _runTestsInTheDirectory(e);
+      }
+    }
+    return testResults;
+  }
+
+  /// Run tests in a single directory under: e2e_tests/web.
+  ///
+  /// Run `flutter pub get` as the first step.
+  ///
+  /// Validate the directory before running the tests. Each directory is
+  /// expected to be a test project which includes a `pubspec.yaml` file
+  /// and a `test_driver` directory.
+  Future<bool> _runTestsInTheDirectory(io.Directory directory) async {
+    final bool directoryContentValid = _validateTestDirectory(directory);
+    if (directoryContentValid) {
+      await _runPubGet(directory.path);
+      final bool testResults = await _runTestsInDirectory(directory);
+      return testResults;
+    } else {
+      return false;
+    }
+  }
+
+  Future<bool> _runTestsInDirectory(io.Directory directory) async {
+    final io.Directory testDirectory =
+        io.Directory(pathlib.join(directory.path, 'test_driver'));
+    final List<io.FileSystemEntity> entities =
+        testDirectory.listSync(followLinks: false);
+
+    final List<String> e2eTestsToRun = List<String>();
+
+    // The following loops over the contents of the directory and saves an
+    // expected driver file name for each e2e test assuming any dart file
+    // not ending with `_test.dart` is an e2e test.
+    // Other files are not considered since developers can add files such as
+    // README.
+    for (io.File f in entities) {
+      final String basename = pathlib.basename(f.path);
+      if (!basename.contains('_test.dart') && basename.endsWith('.dart')) {
+        e2eTestsToRun.add(basename);
+      }
+    }
+
+    print(
```

Comment:
Maybe use Logger class for these with info, warning, error...

Reply:
I think this is good feedback overall for felt tool. thanks.